CONTRIBUTING.md

```diff
@@ -56,6 +56,35 @@ This project follows the
 [Oracle Java code conventions](http://www.oracle.com/technetwork/java/codeconvtoc-136057.html)
 and the Scala conventions detailed above, preferring the latter.
 
+### Test
```

Comment:
```suggestion
### Testing Your Code
```

I also suggest moving this after `Sign Your Work` since that's missed by most new devs to the project and being closer to the top of the file would help.

There's enough stuff in here about testing that we may want to move it into the docs/dev directory and just link to it from this contributing document under a Testing Your Code header.